overlord/snapstate/snapstate.go

```diff
@@ -861,6 +861,29 @@ func InstallMany(st *state.State, names []string, userID int) ([]string, []*stat
 		return nil, nil, err
 	}
 
+	// check if there is enough disk space for requested snaps and their
+	// prerequisites.
+	snapInfos := make([]*snap.Info, len(installs))
+	for i, sar := range installs {
+		snapInfos[i] = sar.Info
+	}
+	totalSize, err := installSize(st, snapInfos, userID)
+	if err != nil {
+		return nil, nil, err
+	}
+	// require 5Mb extra
+	requiredSpace := totalSize + 5*1024*1024
+	path := dirs.SnapdStateDir(dirs.GlobalRootDir)
+	if err := osutilCheckFreeSpace(path, requiredSpace); err != nil {
+		if _, ok := err.(*osutil.NotEnoughDiskSpaceError); ok {
+			return nil, nil, &ErrInsufficientSpace{
```

Comment:
this should be called InsufficientSpaceError per conventions, also it should probably carray a ChangeKind like ChangeConflictError